Mathlib/Order/Iterate.lean

```diff
@@ -0,0 +1,266 @@
+/-
+Copyright (c) 2020 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+-/
+import Mathlib.Logic.Function.Iterate
+import Mathlib.Order.Monotone
+import Mathlib.Data.Nat.Basic
+
+/-!
+# Inequalities on iterates
+
+In this file we prove some inequalities comparing `f^[n] x` and `g^[n] x` where `f` and `g` are
+two self-maps that commute with each other.
+
+Current selection of inequalities is motivated by formalization of the rotation number of
+a circle homeomorphism.
+-/
+
+
+variable {α β : Type _}
```

Comment:
```suggestion
```
We can use autoimplicits here: https://github.com/leanprover-community/mathlib4/wiki#auto-implicits